### PR TITLE
[FLINK-3541] [Kafka Connector] Clean up workaround in FlinkKafkaConsu…

### PR DESCRIPTION
**Pull request by skyahead**
…mer09

### CODE REVIEW

File: flink-streaming-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer09.java [hunk 1]
```diff
@@ -36,6 +36,7 @@
 import org.apache.kafka.common.PartitionInfo;
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.errors.WakeupException;
++import org.apache.kafka.common.KafkaException;
```
Comment (StephanEwen) [inline flink-streaming-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer09.java]: Extra `+`. See also CI logs...

**Comment by StephanEwen** [comment]
This seems to change the behavior. Are you sure that the current workaround (catching the `NullPointerException`) is not needed any more?

**Comment by rmetzger** [comment]
Apparently https://issues.apache.org/jira/browse/KAFKA-2880 has been fixed with Kafka 0.9.0.1. Still, we should make sure that the multiple (I'd say at least 10) test runs are passing without failure.

I'll look into this once its building.

**Comment by StephanEwen** [comment]
@skyahead Will you be fixing the compile issue of this pull request?

**Comment by skyahead** [comment]
@StephanEwen Sorry for late replying. Did not get much time during the weekend on keyboard :-) .

I was trying to understand the test cases related, to address your feedback regarding 'logic being changed'. 

Will work on this from tonight!

**Comment by skyahead** [comment]
@StephanEwen I changed Kafka version from 0.9.0.1 to 0.9.0.0 and verified that NullPointerException does get caught, and the code retries connecting for 10 times. 

Using 0.9.0.1 however, the NullPointerException does not happen anymore, whereas a TimeoutException is thrown as expected and got caught expected too.

All test cases in Kafka08ITCase (19 cases) and Kafka09ITCase (15 cases) pass in my local environment.

**Comment by StephanEwen** [comment]
@rmetzger Given that the Kafka bug is fixed, can we drop the re-try loop, or was that intended to also cover other cases (like brokers having inconsistent metadata) ?

**Comment by rmetzger** [comment]
Yes, we can drop the retry loop.

**Comment by StephanEwen** [comment]
Okay, will merge this on top of the recent changes to the Kafka consumer...

**Pull request closed without merging**